src/geodesy.jl

```diff
@@ -69,8 +88,9 @@ An array of the same type and shape as the `marker`.
 # References
   Morphological Grayscale Reconstruction in Image Analysis: Applications and Efficient Algorithms IEEE Trans Image Process. 1993;2(2)
 """
-function underbuild(marker::AbstractArray{T,N}, mask::AbstractArray{T,N}, connectivity::AbstractArray{Bool}) where {T<:NumberLike,N}
-    check_image_consistency(mask, marker)
+function underbuild(marker::AbstractArray, mask::AbstractArray, connectivity::AbstractArray{Bool})
+    N = ndims(marker)
+    # check_image_consistency(mask, marker)
```

Comment:
why comment ?

Reply:
In 8f28468, I removed quite a few type annotations to support valid mixed input types. For instance: `hmaxima(rand(1:5, 4, 4), h, 3.0)` was not supported even though it does make sense. Usually, we use loose type annotation only to assist multiple dispatches. If some inputs are invalid, they will usually trigger some lower-level errors, e.g., `min(rand(RGB), rand(RGB))` errors. And if we want to provide an informative error message, we can do eager checks as you did in early commits.

I temporarily removed these lines because `typeof(marker) == typeof(mask)` is an overly strong assumption. For instant, the new `hmaxima` version passes `marker::MappedArray` and `mask::Array`. I'll add back necessary checks in new commits.